cmd/keymasterd/2fa_bootstrapOTP.go

```diff
@@ -111,6 +111,44 @@ func (state *RuntimeState) BootstrapOtpAuthHandler(w http.ResponseWriter,
 	}
 }
 
+func (state *RuntimeState) trySelfServiceGenerateBootstrapOTP(username string,
```

Comment:
I dont like the signature of this function, it should return a bool if successful, otherwise we cannot distingusish sucess from failure... and thus as it is right now there is a race condition where the otp generated but fails to be added to the DB.

Thus on the caller you can do
selfServiceOtpGenerated := state.trySelfServiceGenerateBootstrapOTP.....

I really dislike logic where updates to object are not explicitly communicated.

Reply:
I've changed the function signature. The caller hasn't changed because there isn't really anything it would do differently regardless of the return status.